packages/template-retail-react-app/app/ssr.js

```diff
@@ -432,6 +402,65 @@ const {handler} = runtime.createHandler(options, (app) => {
     app.get('/favicon.ico', runtime.serveStaticFile('static/ico/favicon.ico'))
 
     app.get('/worker.js(.map)?', runtime.serveServiceWorker)
+
+    // Helper function to transform relative icon paths to absolute URLs
+    function transformIconPaths(data, ecomServerHost) {
+        const baseUrl = `https://${ecomServerHost}/on/demandware.static/Sites-Site/-/-/internal`
+        const dataStr = JSON.stringify(data)
+        // Replace all relative icon paths with absolute URLs
+        const transformedStr = dataStr.replace(/"src":\s*"\/icons\//g, `"src":"${baseUrl}/icons/`)
+        return JSON.parse(transformedStr)
+    }
+
+    app.get('/api/payment-metadata', async (req, res) => {
```

Comment:
Last time I've said that we may want to move this custom endpoint elsewhere (into another package). But after some research, actually this is the best place to put it. It does not make sense to have this unique salesforce payment code into something generic like the runtime package, for example. Not every project will need to use it. So yeah, let's keep it in the template.

So in the future, for those customers with an existing PWA Kit project, we'll instruct them to manually add this custom endpoint (plus other code diff).